src/openrct2/rct1/S4Importer.cpp

```diff
@@ -2759,16 +2759,16 @@ namespace RCT1
         dst->seat_rotation = DEFAULT_SEAT_ROTATION;
 
         // Vehicle links (indexes converted later)
-        dst->prev_vehicle_on_ride = src->prev_vehicle_on_ride;
-        dst->next_vehicle_on_ride = src->next_vehicle_on_ride;
-        dst->next_vehicle_on_train = src->next_vehicle_on_train;
+        dst->prev_vehicle_on_ride = EntityId::FromUnderlying(src->prev_vehicle_on_ride);
+        dst->next_vehicle_on_ride = EntityId::FromUnderlying(src->next_vehicle_on_ride);
+        dst->next_vehicle_on_train = EntityId::FromUnderlying(src->next_vehicle_on_train);
 
         // Guests (indexes converted later)
         for (int i = 0; i < 32; i++)
         {
-            uint16_t spriteIndex = src->peep[i];
+            const auto spriteIndex = EntityId::FromUnderlying(src->peep[i]);
             dst->peep[i] = spriteIndex;
-            if (spriteIndex != SPRITE_INDEX_NULL)
+            if (spriteIndex.IsNull())
```

Comment:
@ZehMatt you accidentially inverted the condition here.

Reply:
Good catch.